### ISSUE TITLE
Make .NET 4.7.2 and later support .NET Standard 2.0 (not 4.6.1+)

### ISSUE THREAD

**Issue by gravidThoughts** [NuGet/Home#8376]
Can I configure Nuget client infrastructure to behave like the default Paket configuration?

Per a fantastic [article](https://weblog.west-wind.com/posts/2019/Feb/19/Using-NET-Standard-with-Full-Framework-NET) by Rick Strahl: 

> As a side note a number of people pointed out to me that Paket - which is an alternate package manager for NuGet packages - considers versions of .NET 4.7.1 and older incompatible with .NET Standard so you can't actually install .NET Standard packages for those versions by default.

> Makes sense - Microsoft has acknowledged that using .NET Standard on anything prior to 4.7.1 is not a good idea and I would add using 4.7.1 is not optimal either, but alas it does work with some of the messy hackery described above.

> At the end of the day though it's highly recommended that you don't use .NET Standard with 4.7.1 or earlier. Use 4.7.2 if you want to use .NET Standard.

**Comment by rrelyea** [comment]
@terrajobst, @anangaur - is this a good idea. how important?
yes, mistakes were made in this area...

if so, many ways we could enable.
0) keep same behavior
1) change the default
2) change the default, allow override to have old behavior
3) keep same behavior, allow override to have 4.7.2 and above be NS2.

**Comment by terrajobst** [comment]
Changing the NuGet compat mappings seems like a really bad idea because it would be a breaking change for folks where the current state is working fine. The best we could do is issuing a warning from restore when a .NET Standard 1.5 - 2.0 asset is selected for a project targeting .NET Framework 4.6.1 - 4.7.1. I'd be supportive of the latter.

This would also be in line with [our .NET Standard docs](https://docs.microsoft.com/en-us/dotnet/standard/net-standard#net-implementation-support):

![image](https://user-images.githubusercontent.com/5169960/63472458-0c430c00-c427-11e9-9f06-4a8af9f696bf.png)

**Comment by anangaur** [comment]
So do you mean we should have number 3) from Rob’s proposals?

**Comment by terrajobst** [comment]
No, my suggestion would be to add a new warning ID to NuGet and issue a warning as part of restore. The override behavior for this warning would be the usual suppression mechanism via `<NoWarn>`. Like so:

> NU1702: Package 'packageId' was restored using 'netstandardversion' but .NET Standard 1.5 - 2.0 have compatibility issues for .NET Framework versions before 4.7.2. Consider upgrading your project to .NET Framework 4.7.2 or higher. For more details, see 'FWLink'.

I'd imagine the UX would be similar to [NU1701](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1701).

/cc @ericstj @dsplaisted @onovotny
